src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -837,14 +846,11 @@ public synchronized void complete()
     {
         logger.debug("[Stream #{}] handling Complete message, state = {}", planId(), state);
 
-        if (!isFollower)
+        if (!isFollower) // initiator
         {
-            if (state == State.WAIT_COMPLETE)
-                closeSession(State.COMPLETE);
-            else
-                state(State.WAIT_COMPLETE);
+            initiatorCompleteOrWait();
         }
-        else
+        else // follower
```

Comment:
nit: Perhaps we should just reverse the order of the if/else conditions here (and in a few others place) and avoid the comments.

Reply:
> and avoid the comments.

the order isn't what got me, it was when looking at the code it wasn't always clear what code path hit what... I parsed the if statement each time but fond docs were quicker for me to reason.

I can drop, I just know as a new person reading this code it helped me keep things straight

Reply:
I prefer to keep the comments as well. It makes it easier to read the code